swarmauri/standard/llms/concrete/GeminiToolModel.py

```diff
@@ -2,87 +2,112 @@
 import json
 from typing import List, Literal, Dict, Any
 import google.generativeai as genai
+from google.generativeai.protos import FunctionDeclaration
 from swarmauri.core.typing import SubclassUnion
 
 from swarmauri.standard.messages.base.MessageBase import MessageBase
 from swarmauri.standard.messages.concrete.AgentMessage import AgentMessage
 from swarmauri.standard.messages.concrete.FunctionMessage import FunctionMessage
 from swarmauri.standard.llms.base.LLMBase import LLMBase
-from swarmauri.standard.schema_converters.concrete.GeminiSchemaConverter import GeminiSchemaConverter
+from swarmauri.standard.schema_converters.concrete.GeminiSchemaConverter import (
+    GeminiSchemaConverter,
+)
 import google.generativeai as genai
 
+
 class GeminiToolModel(LLMBase):
     """
     3rd Party's Resources: https://ai.google.dev/api/python/google/generativeai/protos/
     """
+
     api_key: str
-    allowed_models: List[str] = ['gemini-1.5-pro-latest']
+    allowed_models: List[str] = ["gemini-1.5-pro-latest"]
     name: str = "gemini-1.5-pro-latest"
-    type: Literal['GeminiToolModel'] = 'GeminiToolModel'
+    type: Literal["GeminiToolModel"] = "GeminiToolModel"
 
     def _schema_convert_tools(self, tools) -> List[Dict[str, Any]]:
-        return [GeminiSchemaConverter().convert(tools[tool]) for tool in tools]
-
-    def _format_messages(self, messages: List[SubclassUnion[MessageBase]]) -> List[Dict[str, str]]:
+        response = [GeminiSchemaConverter().convert(tools[tool]) for tool in tools]
+        logging.info(response)
+        return self._format_tools(response)
+
+    def _format_tools(
+        self, tools: List[SubclassUnion[FunctionMessage]]
```

Comment:
SubclassUnion is used to handle polymorphisms during run-time. It should work, but its not necessary in this instance.